lib/api_watch_diff.ml

```diff
@@ -1 +1,16 @@
-let diff_interface ~reference:_ ~current:_ = false
+let diff_interface ~reference ~current =
+  let current = Cmi_format.read_cmi current in
+  let reference = Cmi_format.read_cmi reference in
```

Comment:
I think @NathanReb wanted for the `Cmi_format.read_cmi` calls to remain in the `bin/api_diff.ml`. So this function should end up taking the parsed interfaces, rather the strings, and these two lines should be moves back into the `run` function. I didn't explain this well in the issue, sorry!

The motive here is to keep all the I/O operations in the executable, and have the library just handle the core logic.

Reply:
No issues @shonfeder! I have moved the parsing of reference and current files back to ```bin/api_diff.ml```. I hope this is fine now.